src/Sarif.Multitool/Rules/SARIF1021.MissingToolVersion.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Microsoft.Json.Pointer;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+{
+    public class MissingToolVersion : SarifValidationSkimmerBase
+    {
+        private readonly MultiformatMessageString _fullDescription = new MultiformatMessageString
+        {
+            Text = RuleResources.SARIF1021_MissingToolVersion
+        };
+
+        public override MultiformatMessageString FullDescription => _fullDescription;
+
+        public override FailureLevel DefaultLevel => FailureLevel.Warning;
+
+        /// <summary>
+        /// SARIF1021
+        /// </summary>
+        public override string Id => RuleId.MissingToolVersion;
+
+        protected override IEnumerable<string> MessageResourceNames => new string[]
+        {
+            nameof(RuleResources.SARIF1021_Default)
```

Comment:
>SARIF1021_Default [](start = 33, length = 17)

This isn't correct. You don't actually return this string from the rule, do you? Also, this data doesn't declare the actual user-facing string that you *are* returning.

Reply:
This is actually correct. This is the name of a resource string into which `LogResult` (below) formats the message arguments.

---
In reply to: [434826465](https://github.com/microsoft/sarif-sdk/pull/1899#discussion_r434826465) [](ancestors = 434826465)